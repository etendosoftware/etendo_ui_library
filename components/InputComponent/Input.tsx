import React from 'react';
import {
  GestureResponderEvent,
  NativeSyntheticEvent,
  StyleSheet,
  TextInputChangeEventData,
  View,
} from 'react-native';
import {WHITE} from '../../styles/colors';

import {
  InputFieldConfigType,
  InputFieldVariants,
  InputImageType,
  inputVariants,
} from './Input.types';

import InputField from './InputComponents/InputField';
import InputTitle from './InputComponents/InputTitle';

export interface Props {
  titleLabel?: string;
  titleImage?: InputImageType;
  placeholder?: string;
  disabled?: boolean;
  value: string;
  data?: Array<any>;
  onPress?: (event: GestureResponderEvent) => void;
  onPressButton?: (event: GestureResponderEvent) => void;
  onChangeText?: (text: string) => void;
  typeField: InputFieldVariants;
}

const Input: React.FC<Props> = ({
  value,
  titleLabel,
  titleImage,
  placeholder,
  disabled,
  onPress,
  onPressButton,
  onChangeText,
  typeField,
}) => {
  return (
    <View style={styles.inputContainer}>
      <InputTitle title={titleLabel} image={titleImage} />
      <InputField
        disabled={disabled}
        config={inputVariants[typeField].field}
        placeholder={placeholder}
        value={value}
        onPress={onPress}
        onPressButton={onPressButton}
        onChangeText={onChangeText}
      />
    </View>
  );
};

export default Input;

const styles = StyleSheet.create({
  inputContainer: {
    flex: 1,
  },
});
