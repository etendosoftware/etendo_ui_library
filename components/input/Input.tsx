import React from 'react';
import {View} from 'react-native';
import {inputStyleVariants, inputVariants, styles} from './Input.style';
import {InputProps} from './Input.types';
import {InputTitle, InputField, InputHelperText} from './index';

const Input: React.FC<InputProps> = ({
  value,
  titleLabel,
  titleImage,
  helperText,
  placeholder,
  disabled,
  maxLength,
  numberOfLines,
  centerText,
  keyboardType,
  onPress,
  onSubmit,
  onFocus,
  onBlur,
  onChangeText,
  typeField,
  isError,
  fontSize,
  height,
  password,
}) => {
  const stateStyle = () => {
    if (typeField === 'onlyRead') {
      return inputStyleVariants.onlyRead;
    } else if (disabled) {
      return inputStyleVariants.disabled;
    } else if (isError) {
      return inputStyleVariants.destructive;
    } else {
      return inputStyleVariants.default;
    }
  };

  return (
    <View style={styles.inputContainer}>
      <InputTitle
        disabled={disabled}
        title={titleLabel}
        image={titleImage}
        styleTitle={stateStyle().titleStyle}
      />
      <InputField
        disabled={disabled}
        configField={inputVariants[typeField].field}
        styleField={stateStyle().fieldStyle}
        placeholder={placeholder}
        value={value}
        onPress={onPress}
        onSubmit={onSubmit}
        onFocus={onFocus ? onFocus : () => {}}
        onBlur={onBlur ? onBlur : () => {}}
        onChangeText={onChangeText}
        keyboardType={keyboardType}
        maxLength={maxLength}
        numberOfLines={numberOfLines}
        centerText={centerText}
        fontSize={fontSize}
        height={height}
        password={password}
      />
      <InputHelperText
        label={helperText}
        styleHelper={stateStyle().helperStyle}
      />
    </View>
  );
};

export default Input;
