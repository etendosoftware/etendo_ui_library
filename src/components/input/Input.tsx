import React from 'react';
import { View } from 'react-native';
import { inputStyleVariants, inputVariants, styles } from './Input.style';
import { InputProps } from './Input.types';
import { InputTitle, InputField, InputHelperText } from './index';

const Input = ({
  value,
  titleLabel,
  titleImage,
  helperText,
  placeholder,
  placeholderPickerSearch,
  disabled = false,
  maxLength,
  centerText,
  keyboardType,
  displayKey,
  dataPicker,
  backgroundColor,
  showOptionsAmount,
  showSearchInPicker,
  height,
  onPress,
  onSubmit,
  onFocus = () => {},
  onBlur = () => {},
  onChangeText,
  onOptionSelected,
  typeField,
  isError,
}: InputProps) => {
  const stateStyle = () => {
    if (typeField === 'readOnly') {
      return inputStyleVariants.readOnly;
    } else if (getDisabled()) {
      return inputStyleVariants.disabled;
    } else if (isError) {
      return inputStyleVariants.destructive;
    } else {
      return inputStyleVariants.default;
    }
  };

  const getDisabled = (): boolean => {
    if (typeField === 'picker') {
      return !dataPicker?.length || disabled;
    }
    if (typeField === 'readOnly') {
      return false;
    }
    return disabled;
  };

  const showFilterSearch = dataPicker && (showSearchInPicker ?? dataPicker?.length > 16);

  return (
    <View style={styles.inputContainer}>
      <InputTitle
        disabled={getDisabled()}
        titleLabel={titleLabel}
        titleImage={titleImage}
        styleTitle={stateStyle().titleStyle}
      />
      <InputField
        disabled={getDisabled()}
        type={typeField}
        configField={inputVariants[typeField].field}
        styleField={stateStyle().fieldStyle}
        placeholder={placeholder}
        placeholderPickerSearch={placeholderPickerSearch}
        value={value}
        onPress={onPress}
        onSubmit={onSubmit}
        onFocus={onFocus}
        onBlur={onBlur}
        onChangeText={onChangeText}
        onOptionSelected={onOptionSelected}
        showOptionsAmount={showOptionsAmount}
        showSearchInPicker={showFilterSearch}
        keyboardType={keyboardType}
        maxLength={maxLength}
        centerText={centerText}
        dataPicker={dataPicker}
        displayKey={displayKey}
        backgroundColor={backgroundColor}
        height={height}
      />
      <InputHelperText
        label={helperText}
        styleHelper={stateStyle().helperStyle}
      />
    </View>
  );
};

export default Input;
