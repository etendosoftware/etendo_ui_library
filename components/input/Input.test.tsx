import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';

describe('Running Test for Input', () => {
  it('Check Input Error', () => {
    const tree = renderer
      .create(
        <Input
          titleLabel="Test"
          helperText="Only characters"
          placeholder="Write a password"
          disabled={false}
          value="Render Test"
          keyboardType="text"
          onPress={() => {
            console.log('Press Field');
          }}
          onChangeText={text => console.log(text)}
          onFocus={() => {
            console.log('On Focus');
          }}
          onBlur={() => {
            console.log('On Blur');
          }}
          typeField="picker"
          isError={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
