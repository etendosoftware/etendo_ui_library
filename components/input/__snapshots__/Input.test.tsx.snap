// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Test for Input Check Input Error 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "height": 29,
        "justifyContent": "flex-start",
        "marginLeft": 5,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#202452",
          "fontSize": 16,
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Array [
        Array [
          Object {
            "borderColor": "transparent",
            "borderRadius": 10,
            "borderWidth": 5,
          },
          Object {
            "borderColor": "#FEF0EF",
          },
        ],
        Array [
          Object {
            "borderColor": "transparent",
          },
        ],
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#F36A62",
          "borderRadius": 5,
          "borderWidth": 2,
          "display": "flex",
          "flexDirection": "row",
          "height": 50,
          "opacity": 1,
          "overflow": "hidden",
          "paddingRight": 0,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "color": "#202452",
                "display": "flex",
                "flex": 1,
                "fontSize": 14,
                "fontWeight": "500",
                "height": 50,
                "minWidth": 0,
                "paddingLeft": 15,
                "textAlignVertical": "center",
              },
            ],
            Object {},
          ]
        }
      >
        Render Test
      </Text>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 35,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="10"
          bbWidth="10"
          fill="none"
          focusable={false}
          height={10}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "height": 10,
                "width": 10,
              },
              Object {
                "flex": 0,
                "height": 10,
                "width": 10,
              },
            ]
          }
          vbHeight={29}
          vbWidth={41}
          width={10}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              clipRule={0}
              d="M25.134 26.8356C22.869 29.5727 18.6723 29.5727 16.4072 26.8356L2.08173 9.52459C-0.973901 5.83214 1.65231 0.249999 6.44513 0.249999L35.0961 0.25C39.8889 0.25 42.5152 5.83213 39.4595 9.52458L25.134 26.8356Z"
              fill={
                Object {
                  "payload": 4280296530,
                  "type": 0,
                }
              }
              fillRule={0}
              propList={
                Array [
                  "fill",
                  "fillRule",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
        "height": 25,
        "marginLeft": 5,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#808080",
            "fontSize": 14,
          },
          Object {
            "color": "#F36A62",
          },
        ]
      }
    >
      Only characters
    </Text>
  </View>
</View>
`;
